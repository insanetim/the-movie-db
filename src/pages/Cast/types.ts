import { ErrorMsg } from 'src/interfaces/global.interface'
import { IMovieDetailsEx } from 'src/interfaces/movie.interface'

export type CastHookReturn = {
  error: ErrorMsg
  loading: boolean
  movie: IMovieDetailsEx | undefined
  movieSlug: string
}

export type CastRouteParams = {
  movieSlug: string
}
