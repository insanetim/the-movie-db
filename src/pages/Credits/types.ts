import { ErrorMsg } from 'src/interfaces/global.interface'
import { IPersonDetails } from 'src/interfaces/person.interface'

export enum FilterOptions {
  All = 'all',
  Cast = 'cast',
  Crew = 'crew',
}

export type CreditsHookReturn = {
  dataSource: ICredit[]
  error: ErrorMsg
  handleChangeFilter: (filter: FilterOptions) => void
  isLoading: boolean
  person?: IPersonDetails
  personSlug: string
}

export type CreditsRouteParams = {
  personSlug: string
}

export interface ICredit {
  key: string
  movieSlug: string
  posterPath: null | string
  releaseDate: string
  releaseDateTitle: string
  role: string
  title: string
}
