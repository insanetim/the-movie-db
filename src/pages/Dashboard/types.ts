import { Query } from 'src/interfaces/global.interface'

export type DashboardHook = {
  query: Query
}
