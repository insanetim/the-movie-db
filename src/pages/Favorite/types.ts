import { ErrorMsg, HandlePagination } from 'src/interfaces/global.interface'
import { IMovie, IMoviesList } from 'src/interfaces/movie.interface'

export type FavoriteHookReturn = {
  error: ErrorMsg
  handleMovieDelete: (
    event: React.MouseEvent<HTMLSpanElement>,
    movieId: IMovie['id']
  ) => () => Promise<void>
  handlePagination: HandlePagination
  loading: boolean
  movies: IMoviesList | null
}
