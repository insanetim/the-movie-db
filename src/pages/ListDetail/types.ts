import { ErrorMsg, HandlePagination } from 'src/interfaces/global.interface'
import { IListDetail } from 'src/interfaces/list.interface'
import { IMovie } from 'src/interfaces/movie.interface'

export type ListDetailHook = {
  error: ErrorMsg
  handleListDelete: () => () => Promise<void>
  handleMovieDelete: (
    event: React.MouseEvent<HTMLSpanElement>,
    movieId: IMovie['id']
  ) => () => Promise<void>
  handlePagination: HandlePagination
  list: IListDetail | null
  loading: boolean
}
