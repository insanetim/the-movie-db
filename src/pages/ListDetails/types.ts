import { ErrorMsg, HandlePagination } from 'src/interfaces/global.interface'
import { IListDetails } from 'src/interfaces/list.interface'
import { IMovie } from 'src/interfaces/movie.interface'

export type ListDetailsHookReturn = {
  error: ErrorMsg
  handleListDelete: () => () => Promise<void>
  handleMovieDelete: (
    event: React.MouseEvent<HTMLSpanElement>,
    movieId: IMovie['id']
  ) => () => Promise<void>
  handlePagination: HandlePagination
  isLoading: boolean
  list?: IListDetails
}

export type ListDetailsRouteParams = {
  listSlug: string
}
