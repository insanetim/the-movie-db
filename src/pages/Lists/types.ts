import { ErrorMsg, HandlePagination } from 'src/interfaces/global.interface'
import { IListsList } from 'src/interfaces/list.interface'

export type ListsHookReturn = {
  error: ErrorMsg
  handleCreateList: () => () => void
  handlePagination: HandlePagination
  lists: IListsList | null
  loading: boolean
}
