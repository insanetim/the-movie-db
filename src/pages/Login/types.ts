import type { IUserData } from 'src/store/session/types'

export interface LoginHook {
  handleLogIn: (userData: IUserData) => Promise<void>
  loading: boolean
}
