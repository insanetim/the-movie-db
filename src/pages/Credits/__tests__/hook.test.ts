import { act, renderHook } from '@testing-library/react'
import React from 'react'
import { useParams } from 'react-router-dom'
import { mockPersonDetails } from 'src/__mocks__/mockPerson'
import { dispatch } from 'src/__mocks__/react-redux'
import * as personDetailsActions from 'src/store/personDetails/actions'
import * as personDetailsSelectors from 'src/store/personDetails/selectors'

import useContainer from '../hook'
import { FilterOptions } from '../types'

jest.useFakeTimers().setSystemTime(new Date('2020-01-01'))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(() => ({ personSlug: '1234-darth-maul' })),
}))

jest.mock('src/store/personDetails/selectors')

describe('Credits useContainer hook', () => {
  jest
    .spyOn(personDetailsSelectors, 'personDetailsLoadingSelector')
    .mockReturnValue(false)
  jest
    .spyOn(personDetailsSelectors, 'personDetailsErrorSelector')
    .mockReturnValue(null)
  const selectPersonById = jest
    .spyOn(personDetailsSelectors, 'personDetailsSelector')
    .mockReturnValue(mockPersonDetails)
  const setState = jest.fn()
  const useStateSpy = jest.spyOn(React, 'useState')
  useStateSpy.mockImplementation(() => [FilterOptions.All, setState])

  it('should match snapshot', () => {
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should match snapshot with "cast" filter', () => {
    useStateSpy.mockImplementationOnce(() => [FilterOptions.Cast, setState])
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should match snapshot with "crew" filter', () => {
    useStateSpy.mockImplementationOnce(() => [FilterOptions.Crew, setState])
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should match snapshot without params', () => {
    jest.mocked(useParams).mockReturnValueOnce({})
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should check "handleChangeFilter" method', () => {
    const { result } = renderHook(useContainer)

    act(() => {
      result.current.handleChangeFilter(FilterOptions.Cast)
    })

    expect(setState).toHaveBeenCalledWith('cast')
  })

  it('should check "useEffect" method', () => {
    const fetchPersonDetails = jest.spyOn(
      personDetailsActions,
      'fetchPersonDetails'
    )
    selectPersonById.mockReturnValueOnce(undefined as never)
    renderHook(useContainer)

    expect(dispatch).toHaveBeenCalled()
    expect(fetchPersonDetails).toHaveBeenCalledWith(1234)
  })
})
