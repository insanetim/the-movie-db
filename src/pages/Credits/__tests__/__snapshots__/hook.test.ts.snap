// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credits useContainer hook should match snapshot 1`] = `
{
  "dataSource": [
    {
      "key": "1234-Neo",
      "movieSlug": "1234-the-matrix",
      "posterPath": "/image",
      "releaseDate": "1999-01-01",
      "releaseDateTitle": "1999",
      "role": "as Neo",
      "title": "The Matrix",
    },
    {
      "key": "1223-undefined",
      "movieSlug": "1223-the-matrix-reloaded",
      "posterPath": "/image",
      "releaseDate": "2003-01-01",
      "releaseDateTitle": "2003",
      "role": "—",
      "title": "The Matrix Reloaded",
    },
    {
      "key": "1234-Director",
      "movieSlug": "1234-the-matrix",
      "posterPath": "/image",
      "releaseDate": "1999-01-01",
      "releaseDateTitle": "1999",
      "role": "Director",
      "title": "The Matrix",
    },
    {
      "key": "4321-Director",
      "movieSlug": "4321-jupiter-ascending",
      "posterPath": null,
      "releaseDate": "2120-01-01T00:00:00.000Z",
      "releaseDateTitle": "—",
      "role": "Director",
      "title": "Jupiter Ascending",
    },
  ],
  "error": null,
  "handleChangeFilter": [Function],
  "loading": false,
  "person": {
    "biography": "test-biography",
    "birthday": "1999-01-01",
    "external_ids": {
      "facebook_id": "facebook_user",
      "instagram_id": "instagram_user",
      "tiktok_id": "tiktok_user",
      "twitter_id": "twitter_user",
    },
    "gender": 0,
    "id": 1234,
    "known_for_department": "Acting",
    "movie_credits": {
      "cast": [
        {
          "character": "Neo",
          "id": 1234,
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "1999-01-01",
          "title": "The Matrix",
        },
        {
          "character": "",
          "id": 1223,
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "2003-01-01",
          "title": "The Matrix Reloaded",
        },
      ],
      "crew": [
        {
          "department": "Directing",
          "id": 1234,
          "job": "Director",
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "1999-01-01",
          "title": "The Matrix",
        },
        {
          "department": "Directing",
          "id": 4321,
          "job": "Director",
          "popularity": 999,
          "poster_path": null,
          "release_date": "",
          "title": "Jupiter Ascending",
        },
      ],
    },
    "name": "test-person",
    "place_of_birth": "New York, USA",
    "popularity": 100500,
    "profile_path": "/image",
  },
  "personSlug": "1234-darth-maul",
}
`;

exports[`Credits useContainer hook should match snapshot with "cast" filter 1`] = `
{
  "dataSource": [
    {
      "key": "1234-Neo",
      "movieSlug": "1234-the-matrix",
      "posterPath": "/image",
      "releaseDate": "1999-01-01",
      "releaseDateTitle": "1999",
      "role": "as Neo",
      "title": "The Matrix",
    },
    {
      "key": "1223-undefined",
      "movieSlug": "1223-the-matrix-reloaded",
      "posterPath": "/image",
      "releaseDate": "2003-01-01",
      "releaseDateTitle": "2003",
      "role": "—",
      "title": "The Matrix Reloaded",
    },
  ],
  "error": null,
  "handleChangeFilter": [Function],
  "loading": false,
  "person": {
    "biography": "test-biography",
    "birthday": "1999-01-01",
    "external_ids": {
      "facebook_id": "facebook_user",
      "instagram_id": "instagram_user",
      "tiktok_id": "tiktok_user",
      "twitter_id": "twitter_user",
    },
    "gender": 0,
    "id": 1234,
    "known_for_department": "Acting",
    "movie_credits": {
      "cast": [
        {
          "character": "Neo",
          "id": 1234,
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "1999-01-01",
          "title": "The Matrix",
        },
        {
          "character": "",
          "id": 1223,
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "2003-01-01",
          "title": "The Matrix Reloaded",
        },
      ],
      "crew": [
        {
          "department": "Directing",
          "id": 1234,
          "job": "Director",
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "1999-01-01",
          "title": "The Matrix",
        },
        {
          "department": "Directing",
          "id": 4321,
          "job": "Director",
          "popularity": 999,
          "poster_path": null,
          "release_date": "",
          "title": "Jupiter Ascending",
        },
      ],
    },
    "name": "test-person",
    "place_of_birth": "New York, USA",
    "popularity": 100500,
    "profile_path": "/image",
  },
  "personSlug": "1234-darth-maul",
}
`;

exports[`Credits useContainer hook should match snapshot with "crew" filter 1`] = `
{
  "dataSource": [
    {
      "key": "1234-Director",
      "movieSlug": "1234-the-matrix",
      "posterPath": "/image",
      "releaseDate": "1999-01-01",
      "releaseDateTitle": "1999",
      "role": "Director",
      "title": "The Matrix",
    },
    {
      "key": "4321-Director",
      "movieSlug": "4321-jupiter-ascending",
      "posterPath": null,
      "releaseDate": "2120-01-01T00:00:00.000Z",
      "releaseDateTitle": "—",
      "role": "Director",
      "title": "Jupiter Ascending",
    },
  ],
  "error": null,
  "handleChangeFilter": [Function],
  "loading": false,
  "person": {
    "biography": "test-biography",
    "birthday": "1999-01-01",
    "external_ids": {
      "facebook_id": "facebook_user",
      "instagram_id": "instagram_user",
      "tiktok_id": "tiktok_user",
      "twitter_id": "twitter_user",
    },
    "gender": 0,
    "id": 1234,
    "known_for_department": "Acting",
    "movie_credits": {
      "cast": [
        {
          "character": "Neo",
          "id": 1234,
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "1999-01-01",
          "title": "The Matrix",
        },
        {
          "character": "",
          "id": 1223,
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "2003-01-01",
          "title": "The Matrix Reloaded",
        },
      ],
      "crew": [
        {
          "department": "Directing",
          "id": 1234,
          "job": "Director",
          "popularity": 100500,
          "poster_path": "/image",
          "release_date": "1999-01-01",
          "title": "The Matrix",
        },
        {
          "department": "Directing",
          "id": 4321,
          "job": "Director",
          "popularity": 999,
          "poster_path": null,
          "release_date": "",
          "title": "Jupiter Ascending",
        },
      ],
    },
    "name": "test-person",
    "place_of_birth": "New York, USA",
    "popularity": 100500,
    "profile_path": "/image",
  },
  "personSlug": "1234-darth-maul",
}
`;
